### PR TITLE
feat(company-info): implement new settings dropdown design

### PR DESCRIPTION
**Pull request by goldenmaya**
Description:
https://virtocommerce.atlassian.net/browse/ST-4661

--

QA-test:

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.39.0-pr-720-fcb2-fcb2ca3f.zip

### CODE REVIEW

File: client-app/pages/company/info.vue [hunk 1]
```diff
@@ -195,27 +220,24 @@
...
-                        :class="['mr-1', { 'text-[--color-danger-500]': !address.isDefault }]"
-                        name="x"
-                        :size="18"
-                      />
-                      <span class="text-sm">{{ $t("common.buttons.delete") }}</span>
+                      <VcIcon :class="{ 'text-[--color-danger-500]': !address.isDefault }" name="delete-2" size="sm" />
+
+                      {{ $t("common.buttons.delete") }}
                     </button>
                   </VcActionDropdownMenu>
```
Comment (asvishnyakov) [inline client-app/pages/company/info.vue:242]: Reuse this code and don't forget about permissions checking

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/pages/company/info.vue [hunk 2]
```diff
@@ -76,7 +76,6 @@
         <div v-else class="flex flex-col md:rounded md:border">
           <VcTable
             :loading="loadingAddresses"
-            :item-actions-builder="itemActionsBuilder"
```
Comment (asvishnyakov) [inline client-app/pages/company/info.vue:79]: Remove no longer used variable `itemActionsBuilder` from typescript

File: tailwind.config.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { BREAKPOINTS } from "./client-app/core/constants/tailwind";
 import type { Config } from "tailwindcss";
+import {deprecate} from "util";
```
Comment (asvishnyakov) [inline tailwind.config.ts:3]: You're using jsdoc `@deprecated` and don't need to import this node.js function

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

**Review by asvishnyakov — CHANGES_REQUESTED** [review]
That's definitely not that I mean when I asked for reused code. I simply mean you should create shared component for that VcActionDropDownMenu with it's items and then use it twice in code. I think refactoring of the VcActionDropDownMenu should be part of UI Kit work and it should be done in the other way, like https://vuetifyjs.com/en/components/menus/ so you can be able to use component-based approach to build the menu, instead of passing classes, event handlers and other stuff as props. So it should be a separate task with mitigation about UI kit first, especially because we have the same actions on other pages and also other component which using VcActionDropDownMenu

**Review by NaMax66 — APPROVED** [review]

File: client-app/ui-kit/components/atoms/action-dropdown-menu/vc-action-dropdown-menu.vue [hunk 4]
```diff
@@ -1,3 +1,4 @@
+<!-- @deprecated the component will be removed from ui-kit -->
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/atoms/action-dropdown-menu/vc-action-dropdown-menu.vue:1]: https://github.com/VirtoCommerce/vc-theme-b2b-vue/blob/c31ea77061d7407bf6b1cb385c37c7a8c190061c/client-app/ui-kit/components/atoms/types.d.ts#L7
Put it here instead please

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**